File: v131016/TeeNew/Gradient_MidColor.pas
unit Gradient_MidColor;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, TeePenDlg;

type
  TGradientMiddle = class(TBaseForm)
    Label1: TLabel;
    Shape1: TShape;
    CheckBox1: TCheckBox;
    procedure Shape1MouseUp(Sender: TObject; Button: TMouseButton;
      Shift: TShiftState; X, Y: Integer);
    procedure FormCreate(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeCanvas;

procedure TGradientMiddle.Shape1MouseUp(Sender: TObject; Button: TMouseButton;
  Shift: TShiftState; X, Y: Integer);
begin
  inherited;

  With Chart1.Gradient do
  begin
    MidColor:=EditColor(Self,MidColor);
    Shape1.Brush.Color:=MidColor;
    CheckBox1.Checked:=Chart1.Gradient.MidColor=clNone;
  end;
end;

procedure TGradientMiddle.FormCreate(Sender: TObject);
begin
  inherited;
  Shape1.Brush.Color:=Chart1.Gradient.MidColor;
end;

procedure TGradientMiddle.CheckBox1Click(Sender: TObject);
begin
  inherited;

  if CheckBox1.Checked then Chart1.Gradient.MidColor:=clNone
                       else Chart1.Gradient.UseMiddleColor;
end;

initialization
  RegisterClass(TGradientMiddle);
end.

File: TeeNew/Builder2007/Gradient_MidColor.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <TeCanvas.hpp>
#pragma hdrstop

#include "Gradient_MidColor.h"

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TGradientMiddle *GradientMiddle;
//---------------------------------------------------------------------------
__fastcall TGradientMiddle::TGradientMiddle(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TGradientMiddle::FormCreate(TObject *Sender)
{
  Shape1->Brush->Color =Chart1->Gradient->MidColor;
}
//---------------------------------------------------------------------------

void __fastcall TGradientMiddle::Shape1MouseUp(TObject *Sender,
      TMouseButton Button, TShiftState Shift, int X, int Y)
{
  Chart1->Gradient->MidColor = EditColor(this,Chart1->Gradient->MidColor);
  Shape1->Brush->Color = Chart1->Gradient->MidColor;
}
//---------------------------------------------------------------------------

void __fastcall TGradientMiddle::CheckBox1Click(TObject *Sender)
{
  Shape1->Visible != CheckBox1->Checked;
  if (CheckBox1->Checked) Chart1->Gradient->MidColor = clNone;
  else Chart1->Gradient->MidColor = Shape1->Brush->Color;
}
//---------------------------------------------------------------------------

